Frequent test fail assign_mission_target_needs_om_special

### Describe the bug

```
Run ./tools/json_tools/assign_mission_target_needs_om_special.bash [2](https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/7463239515/job/20307441980?pr=70801#step:3:2)>&1
  ./tools/json_tools/assign_mission_target_needs_om_special.bash 2>&1
  shell: /usr/bin/bash -e {0}
jq: error (at ./data/json/overmap/overmap_special/nuclear_plant.json:602): null (null) cannot be matched, as it is not a string
jq: error (at ./data/json/overmap/overmap_special/specials.json:7[3](https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/7463239515/job/20307441980?pr=70801#step:3:3)9[4](https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/7463239515/job/20307441980?pr=70801#step:3:5)): null (null) cannot be matched, as it is not a string
jq: error (at ./data/json/overmap/overmap_special/overmap_special_vitrified.json:2[5](https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/7463239515/job/20307441980?pr=70801#step:3:6)): null (null) cannot be matched, as it is not a string
jq: error (at ./data/json/overmap/overmap_special/lab_modular/lab_mutagen_[6](https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/7463239515/job/20307441980?pr=70801#step:3:7)_floor.json:13[7](https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/7463239515/job/20307441980?pr=70801#step:3:8)): null (null) cannot be matched, as it is not a string
jq: error (at ./data/json/overmap/overmap_special/mine.json:[10](https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/7463239515/job/20307441980?pr=70801#step:3:11)5): null (null) cannot be matched, as it is not a string
jq: error (at ./data/json/mapgen/map_extras/riverside_boats.json:154): Cannot index string with string "type"
jq: error (at ./data/mods/TEST_DATA/vehicle_export_test.json:35): Cannot index string with string "type"
jq: error (at ./data/json/mapgen/map_extras/riverside_boats.json:154): Cannot index string with string "type"
jq: error (at ./data/mods/TEST_DATA/vehicle_export_test.json:35): Cannot index string with string "type"
The following missions lack .start.assign_mission_target.om_special
See https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/MISSIONS_JSON.md#assign_mission_target
> If the om_terrain is part of an overmap special, it's essential to specify
> the om_special value as well--otherwise, the game will not know how to spawn
> the entire special.
[
  {
    "id": "MISSION_EXPLORE_SARCOPHAGUS",
    "om_special": "Hazardous Waste Sarcophagus",
    "XXX_NEEDS_SPECIAL": [
      "nuclear power plant"
    ]
  }
]
Error: Process completed with exit code 1.
```
I don't really get what the error means, the mission def:
https://github.com/CleverRaven/Cataclysm-DDA/blob/72d5293688cbe13377247c2ade6f5dc2aa1abe9f/data/json/npcs/missiondef.json#L2-L17
Defines a valid special that spawns with the target `om_terrain`:
https://github.com/CleverRaven/Cataclysm-DDA/blob/72d5293688cbe13377247c2ade6f5dc2aa1abe9f/data/json/overmap/overmap_special/specials.json#L1880-L1900
but is insisting in needs to be "nuclear power plant", a different special that has two sarcophagi with the same `om_terrain`s:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/overmap/overmap_special/nuclear_plant.json

The `om_special` field in the mission def doesn't appear to accept an array but can be omitted altogether, idk if the nuclear power plant is a sensible target for this mission though

### Attach save file

N/A

### Steps to reproduce

N/A

### Expected behavior

Tests not to fail

### Screenshots

_No response_

### Versions and configuration

N/A

### Additional context

_No response_

--- a/tools/json_tools/assign_mission_target_needs_om_special.bash
+++ b/tools/json_tools/assign_mission_target_needs_om_special.bash
@@ -15,6 +15,7 @@ Q 'def skip(to_skip): select(all(. != to_skip; .));
 	| select(.type=="overmap_special")
 	| .id as $id
 	| .overmaps[].overmap
+    | select(. != null)
 	| sub("_(north|south|east|west)$"; "")
 	# Skip common terrains to prevent false positives
 	| skip("forest", "forest_thick", "forest_water", "field", "road_ew")

